pkg/kubeadm/util/tokens.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
+)
+
+const (
+	TokenIDLen = 6
+	TokenBytes = 8
+)
+
+func randBytes(length int) ([]byte, string, error) {
+	b := make([]byte, length)
+	_, err := rand.Read(b)
```

Comment:
This one is crypto/rand actually, which is used on Joe's advice.

On Thu, 8 Sep 2016, 01:07 Mike Danese, <notifications@github.com> wrote:

> In pkg/kubeadm/util/tokens.go
> <https://github.com/kubernetes/kubernetes/pull/31221#discussion_r77927230>
> :
>
> > +	"crypto/rand"
> > +	"encoding/hex"
> > +	"fmt"
> > +	"strings"
> > +
> > +	kubeadmapi "k8s.io/kubernetes/pkg/kubeadm/api"
> > +)
> > +
> > +const (
> > +	TokenIDLen = 6
> > +	TokenBytes = 8
> > +)
> > +
> > +func randBytes(length int) ([]byte, string, error) {
> > +	b := make([]byte, length)
> > +	_, err := rand.Read(b)
>
> Use k8s.io/kubernetes/pkg/util/rand since it's threadsafe and rand is not
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31221/files/6e0fe5561364cdd1a098268ea6af7d78d418ffe4#r77927230>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPWS4XLXBo_8xYEtEZr98mvu8PnPDsrks5qn1GtgaJpZM4JqsTH>
> .
>